core/src/main/java/org/kiwix/kiwixmobile/core/search/viewmodel/effects/ShowDeleteSearchDialog.kt

```diff
@@ -36,7 +35,6 @@ data class ShowDeleteSearchDialog(
   @Inject lateinit var dialogShower: DialogShower
 
   override fun invokeWith(activity: AppCompatActivity) {
-    (activity as SearchActivity).activityComponent.inject(this)
```

Comment:
This needs the injection, it is why I retrieved the activity component from the activity. I can explain in more depth if you want